src/middlewared/middlewared/plugins/usage.py

```diff
@@ -0,0 +1,416 @@
+        lags = await gather_lags()
+        phys = await gather_physical()
+        vlans = await gather_vlans()
+
+        return {'network': {**bridges, **lags, **phys, **vlans}}
+
+    @private
+    async def gather_system(self):
+        system = await self.middleware.call('system.info')
+        platform = 'FreeNAS' if await self.middleware.call(
+            'system.is_freenas'
+        ) else 'TrueNAS'
+
+        usage_version = 1
+        version = system['version']
+        datasets = await self.middleware.call(
+            'zfs.dataset.query', [('type', '!=', 'VOLUME')], {'count': True}
+        )
+        users = await self.middleware.call(
+            'user.query', [], {'count': True}
+        )
+        snapshots = await self.middleware.call(
+            'zfs.snapshot.query', [], {'count': True}
+        )
+        zvols = await self.middleware.call(
+            'zfs.dataset.query', [('type', '=', 'VOLUME')], {'count': True}
+        )
+
+        return {
+            'gather_system': [
+                {'platform': platform},
+                {'usage_version': usage_version},
+                {'version': version},
+                {'system': [
+                    {
+                        'users': users, 'snapshots': snapshots, 'zvols': zvols,
+                        'datasets': datasets
+                    }
+                ]}
+            ]
+        }
+
+    @private
+    async def gather_plugins(self):
+        plugins = await self.middleware.call('jail.list_resource', 'PLUGIN')
+        plugin_list = []
+
+        for p in plugins:
+            plugin_list.append(
+                {
+                    'name': p[1],
+                    'version': p[10]
+                }
+            )
+
+        return {'plugins': plugin_list}
+
+    @private
+    async def gather_pools(self):
+        pools = await self.middleware.call('pool.query')
+        pool_list = []
+
+        for p in pools:
+            if p['status'] == 'OFFLINE':
+                continue
+
+            disks = 0
+            vdevs = 0
+            type = 'STRIPE'
+
+            try:
+                pd = (await self.middleware.call(
+                    'zfs.dataset.query', [('id', '=', p['name'])],
+                    {'get': True}
+                ))['properties']
+            except IndexError:
+                self.logger.error(
+                    f'{p["name"]} is missing, skipping collection',
+                    exc_info=True
+                )
+
+            for d in p['topology']['data']:
+                if not d.get('path'):
+                    vdevs += 1
+                    type = d['type']
+                    disks += len(d['children'])
+                else:
+                    disks += 1
+                    used = pd['usedbysnapshots']['parsed'] + \
+                        pd['usedbydataset']['parsed'] + \
+                        pd['usedbychildren']['parsed'] + \
+                        pd['usedbyrefreservation']['parsed']
```

Comment:
I think I was not clear here,

this is the same as just `pd['usedbydataset']['parsed']`

Thought we could perhaps also send other usedby* to usage stats.